components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1263,6 +1263,10 @@ void RewardsServiceImpl::OnStopEngineForCompleteReset(
     const mojom::Result result) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   profile_->GetPrefs()->ClearPrefsWithPrefixSilently(pref_prefix);
+  // Notify `brave_rewards::prefs::kEnabled` listeners that Brave Rewards was
+  // disabled.
+  profile_->GetPrefs()->SetBoolean(brave_rewards::prefs::kEnabled, false);
```

Comment:
Setting this value to a non-default value may cause the Rewards engine utility process to always start (see line 428). We definitely want to avoid that.

It might be enough to simply change the condition on line 428 to test the value of the "enabled" pref, rather than testing for a non-default value, but please test to make sure that the Rewards engine process starts and stops appropriately.

Reply:
Also, it's unfortunate that we can't simply clear prefs and be done. What part of the system are we trying to notify by setting the pref here?

Reply:
This change covers an edge case where the "Delete browsing data" modal would need to be on the screen at the same time Rewards was reset. This is an edge case and not worth the risk, or amount of effort for regression testing. I would recommend this change is removed because once the modal is closed and reopened the correct text would be shown.

Reply:
Agree that this is an edge case. Removed this setting pref change above:
`profile_->GetPrefs()->SetBoolean(brave_rewards::prefs::kEnabled, false);`